packages/realm-server/billing/stripe-webhook-handlers/subscribe.ts

```diff
@@ -77,3 +78,22 @@ export async function handlePaymentSucceeded(
 
   return new Response('ok');
 }
+
+export async function handleCheckoutSessionCompleted(
+  dbAdapter: DBAdapter,
+  event: StripeEvent,
+): Promise<Response> {
+  // Scenario 1: a user completes initial subscription through checkout session
+  // Scenario 2: a user resubscribes to a plan or upgrades/downgrades their plan
+
+  // TODO: handle scenario 2 to not create new user if user already exists
```

Comment:
@jurgenwerk Based on the [Checkout Lifecycle](https://docs.stripe.com/payments/checkout/how-checkout-works?payment-ui=stripe-hosted&locale=en-GB#lifecycle), checkout can occur not only during an initial subscription but also when a user:

- Resubscribes to a plan
- Upgrades or downgrades their plan

I think we shall also do a check to skip `INSERT` query to `user` table if the record already exists. What do you think?

Reply:
Not sure I follow–the user should already be present in the users table at this point. We want to update the user by saving their stripe's customer id, in case they don't have it yet

Reply:
I see @jurgenwerk, I noticed that in the latest migration file about Stripe. I shall do the update query instead